src/kml/src/test/java/org/geoserver/kml/icons/IconPropertiesTest.java

```diff
@@ -83,6 +85,20 @@ public void testDynamicMark() throws CQLException {
         assertEquals("0.0.0=&0.0.0.name=circle", encode(s, fieldIs1));
         assertEquals("0.0.0=&0.0.0.name=square", encode(s, fieldIs2));
     }
+    
+    @Test
+    public void testDynamicColor() throws CQLException {
+        Expression color = toExpression("if_then_else(equalTo(field, 1), '#8080C0', '#CC8030')");
+        Stroke stroke = styleFactory.stroke(color,  null,  null, null, null, null, null);
+        Fill fill = styleFactory.fill(null, color, null );        
+        Mark mark = styleFactory.mark(toExpression("circle"), fill, stroke);
+        Graphic graphic = styleFactory.graphic( Collections.singletonList(mark),null, null, null, null, null);       
+        PointSymbolizer symbolizer = styleFactory.pointSymbolizer("symbolizer", toExpression("geom"),  null,  null, graphic );
+        
+        final Style s = styleFromRules(catchAllRule(symbolizer));
+        assertEquals("0.0.0=&0.0.0.fill.color=%238080C0&0.0.0.name=&0.0.0.stroke.color=%238080C0", encode(s, fieldIs1));
+        assertEquals("0.0.0=&0.0.0.fill.color=C08030&0.0.0.name=&0.0.0.stroke.color=C08030", encode(s, fieldIs2));
```

Comment:
The above encoding shows RGB hex, expect we should be using ABGR order hex.

Reply:
Hmm... don't know honestly? But it's failing:

{code}
Failed tests:   testDynamicColor(org.geoserver.kml.icons.IconPropertiesTest): expected:<...0=&0.0.0.fill.color=[C08030&0.0.0.name=&0.0.0.stroke.color=C0]8030> but was:<...0=&0.0.0.fill.color=[%23CC8030&0.0.0.name=&0.0.0.stroke.color=%23CC]8030>
{code}

Check the icon server expectations (was not written by me)